entry_types/scrolled/package/spec/entryState/structure-spec.js

```diff
@@ -199,11 +199,42 @@ describe('useChapters', () => {
     expect(chapters).toMatchObject([
       {
         permaId: 10,
+        chapterRef: 'chapter-1',
         title: 'Chapter 1',
         summary: 'An introductory chapter'
       },
       {
         permaId: 11,
+        chapterRef: 'chapter-2',
+        title: 'Chapter 2',
+        summary: 'A great chapter'
+      }
+    ]);
+  });
+
+  it('escapes chapter titles', () => {
+    chaptersSeed[0].configuration.title = "<script>alert('test')</script>Titleä"
+    const {result} = renderHookInEntry(
+      () => useChapters(),
+      {
+        seed: {
+          chapters: chaptersSeed
+        }
+      }
+    );
+
+    const chapters = result.current;
+
+    expect(chapters).toMatchObject([
+      {
+        permaId: 10,
+        chapterRef: "lessscriptgreateralert('test')lessscriptgreatertitlea",
+        title: chaptersSeed[0].configuration.title,
+        summary: 'An introductory chapter'
+      },
+      {
+        permaId: 11,
+        chapterRef: 'chapter-2',
```

Comment:
**suggestion**: Use separate seed only for this test

If we used an array of chapter seed data directly inside of this test, we could include more different examples. Since the passed [seed data is normalized](https://github.com/codevise/pageflow/blob/cf0fe63d1804eeb643530a906a4aef9020229b7b/entry_types/scrolled/package/spec/support/normalizeSeed.js#L131), we would only need to include the chapter titles and could skip all other unrelated data like `permaId` or `id`. This would make it easy to add additional examples in the future and also allow covering currently untested logic like making chapter slugs unique.

It also [looks like Slugify supports replacing umlauts](https://github.com/simov/slugify/blob/d2646e97cf1bb3784b38dbf7ca4a45b1a925b110/slugify.js#L16) if we pass the `locale` option. I'd also consider enabling the `strict` option to remove all other potentially problematic characters.

We could then replace the above "<script>" example, with more realistic examples for things like whitespace or special characters. As we discussed, script injection is not the vector we are addressing here. Therefore this test example could be confusing someone trying to understand this part of the code base.

Reply:
**suggestion**: Add slug property to expected result in `useEntryStructure` test

Since we depend on the slug to also be included in the data returned by `useEntryStructure` and actually changed the implementation of `useEntryStructure` to ensure this, we should also change the test. Since we know that all the details of escaping chapter slugs are already covered by the `useChapters` test, I'd suggest to use the existing test and only add the slug property to `expectedEntryStructure`.

Reply:
tried both locale en and de but umlauts are not being translated like ä->ae.

Reply:
strict is also not working as expected I think because $ is replaced by dollar. I expected strict to remove it
also <script> translates to lessscriptgreater in strict mode.

Reply:
see [comment below](https://github.com/codevise/pageflow/pull/1637#discussion_r538460783).